preditor/gui/group_tab_widget/one_tab_widget.py

```diff
@@ -17,6 +17,42 @@ def __init__(self, *args, **kwargs):
         super(OneTabWidget, self).__init__(*args, **kwargs)
         self.tabCloseRequested.connect(self.close_tab)
 
+    def get_next_available_tab_name(self, name):
+        """Get the next available tab name, incrementing an iteration if needed.
+
+        Args:
+            name (str): The desired name
+
+        Returns:
+            name (str): The name, or updated name if needed
+        """
+        name = name.replace(" ", "_")
+
+        existing_names = [self.tabText(i) for i in range(self.count())]
+
+        base = ""
```

Comment:
I would like to see more comments on this explaining what each step is doing. Ie the first for loop is finding the last non-digit character, etc.